homeassistant/components/broadlink/switch.py

```diff
@@ -110,58 +109,53 @@ def get_mp1_slot_name(switch_friendly_name, slot):
         return slots[f"slot_{slot}"]
 
     if model in RM_TYPES:
-        broadlink_device = blk.rm((host, DEFAULT_PORT), mac_addr, None)
-        hass.add_job(async_setup_service, hass, host, broadlink_device)
+        api = blk.rm((host, DEFAULT_PORT), mac_addr, None)
+        broadlink_device = BroadlinkDevice(hass, api)
+        hass.async_create_task(async_setup_service(hass, host, broadlink_device))
```

Comment:
Why do we need to create a task instead of awaiting `async_setup_service`?

Reply:
Should we really register the device for the service before we know that we can finish platform setup?

Reply:
The idea behind creating a task is that we don't have to wait until it is completed, as this platform does not depend on these services. But you're right, it needs to be done in the end, so we don't risk registering a device that is offline. I'll fix it. Thank you.

Reply:
Done.